docs/architecture/director-component.md

```diff
@@ -0,0 +1,42 @@
+# Director Component
+
+## Overview
+
+This document describes [Director](../terminology.md)'s architecture.
+
+## Architecture
+
+The following diagram illustrates the overall architecture of this application:
```

Comment:
I have some doubt whether the services and repositories should be drawn as a part of Root Resolver. In general, those components should not be aware that the layer which calls them is a GraphQL or anything else.

Reply:
Hmm, technically it's composition and that's what I wanted to show:
Root Resolver consists of Domain Resolvers, Domain Resolvers consists of Services, and Services of Repositories. 🤔

Reply:
Maybe just do not enclose it in one block

Reply:
@crabtree What about now?

Reply:
👍